### PR TITLE
timer_management: Get numeric timezone automatically and a bug fix

### PR DESCRIPTION
**Pull request by will-Do**
1). Get numeric timezone automatically as some timezones may have
use DST.
2). Rectify the expected result between timezone and UTC time in VM.

Signed-off-by: Yanbing Du <ydu@redhat.com>

### CODE REVIEW

**Comment by will-Do** [comment]
All pass on libvirt-1.2.9.1-2.fc20.x86_64
```
  ./run -t libvirt --test  ydu-libvirt.timer_management -k --keep-image-between-tests
SETUP: PASS (0.60 s)
DATA DIR: /home/ydu/Work/virt-test/shared/data
DEBUG LOG: /home/ydu/Work/virt-test/logs/run-2015-04-08-10.40.13/debug.log
TESTS: 83
(1/83) type_specific.ydu-libvirt.timer_management.no_timer.offset_utc.under_stress.stress_in_vm.memory.test_timers_in_vm.linux_guest: PASS (60.69 s)
(2/83) type_specific.ydu-libvirt.timer_management.no_timer.offset_utc.under_stress.stress_in_vm.vcpu.test_timers_in_vm.linux_guest: PASS (61.58 s)
(3/83) type_specific.ydu-libvirt.timer_management.no_timer.offset_utc.under_stress.stress_on_host.test_timers_in_vm.linux_guest: PASS (60.81 s)
(4/83) type_specific.ydu-libvirt.timer_management.no_timer.offset_utc.under_stress.vm_operations.inject_nmi.test_timers_in_vm.linux_guest: PASS (61.23 s)
(5/83) type_specific.ydu-libvirt.timer_management.no_timer.offset_utc.under_stress.vm_operations.dump.test_timers_in_vm.linux_guest: PASS (83.26 s)
(6/83) type_specific.ydu-libvirt.timer_management.no_timer.offset_utc.under_stress.vm_operations.suspend_resume.test_timers_in_vm.linux_guest: PASS (61.46 s)
(7/83) type_specific.ydu-libvirt.timer_management.no_timer.offset_utc.under_stress.vm_operations.save_restore.test_timers_in_vm.linux_guest: PASS (111.96 s)
(8/83) type_specific.ydu-libvirt.timer_management.no_timer.offset_utc.no_stress.test_timers_in_vm.linux_guest: PASS (62.33 s)
(9/83) type_specific.ydu-libvirt.timer_management.no_timer.offset_timezone.no_stress.test_timers_in_vm.linux_guest: PASS (62.04 s)
(10/83) type_specific.ydu-libvirt.timer_management.no_timer.offset_variable.no_stress.test_timers_in_vm.linux_guest: PASS (60.93 s)
(11/83) type_specific.ydu-libvirt.timer_management.no_timer.offset_localtime.no_stress.test_timers_in_vm.linux_guest: PASS (60.79 s)
(12/83) type_specific.ydu-libvirt.timer_management.specific_timer.kvmclock.present_no.offset_utc.no_stress.test_specific_timer: PASS (55.28 s)
(13/83) type_specific.ydu-libvirt.timer_management.specific_timer.kvmclock.present_no.offset_timezone.no_stress.test_specific_timer: PASS (55.30 s)
(14/83) type_specific.ydu-libvirt.timer_management.specific_timer.kvmclock.present_no.offset_variable.no_stress.test_specific_timer: PASS (55.92 s)
(15/83) type_specific.ydu-libvirt.timer_management.specific_timer.kvmclock.present_no.offset_localtime.no_stress.test_specific_timer: PASS (55.09 s)
(16/83) type_specific.ydu-libvirt.timer_management.specific_timer.kvmclock.present_yes.offset_utc.no_stress.test_specific_timer: PASS (55.89 s)
(17/83) type_specific.ydu-libvirt.timer_management.specific_timer.kvmclock.present_yes.offset_timezone.no_stress.test_specific_timer: PASS (57.97 s)
(18/83) type_specific.ydu-libvirt.timer_management.specific_timer.kvmclock.present_yes.offset_variable.no_stress.test_specific_timer: PASS (61.51 s)
(19/83) type_specific.ydu-libvirt.timer_management.specific_timer.kvmclock.present_yes.offset_localtime.no_stress.test_specific_timer: PASS (93.66 s)
(20/83) type_specific.ydu-libvirt.timer_management.specific_timer.hpet.present_no.offset_utc.no_stress.test_specific_timer: PASS (69.75 s)
(21/83) type_specific.ydu-libvirt.timer_management.specific_timer.hpet.present_no.offset_timezone.no_stress.test_specific_timer: PASS (49.87 s)
(22/83) type_specific.ydu-libvirt.timer_management.specific_timer.hpet.present_no.offset_variable.no_stress.test_specific_timer: PASS (49.01 s)
(23/83) type_specific.ydu-libvirt.timer_management.specific_timer.hpet.present_no.offset_localtime.no_stress.test_specific_timer: PASS (49.24 s)
(24/83) type_specific.ydu-libvirt.timer_management.specific_timer.hpet.present_yes.offset_utc.no_stress.test_specific_timer: PASS (51.81 s)
(25/83) type_specific.ydu-libvirt.timer_management.specific_timer.hpet.present_yes.offset_timezone.no_stress.test_specific_timer: PASS (51.94 s)
(26/83) type_specific.ydu-libvirt.timer_management.specific_timer.hpet.present_yes.offset_variable.no_stress.test_specific_timer: PASS (51.81 s)
(27/83) type_specific.ydu-libvirt.timer_management.specific_timer.hpet.present_yes.offset_localtime.no_stress.test_specific_timer: PASS (51.76 s)
(28/83) type_specific.ydu-libvirt.timer_management.specific_timer.pit.present_no.offset_utc.no_stress.test_specific_timer: PASS (50.88 s)
(29/83) type_specific.ydu-libvirt.timer_management.specific_timer.pit.present_no.offset_timezone.no_stress.test_specific_timer: PASS (52.10 s)
(30/83) type_specific.ydu-libvirt.timer_management.specific_timer.pit.present_no.offset_variable.no_stress.test_specific_timer: PASS (53.07 s)
(31/83) type_specific.ydu-libvirt.timer_management.specific_timer.pit.present_no.offset_localtime.no_stress.test_specific_timer: PASS (51.97 s)
(32/83) type_specific.ydu-libvirt.timer_management.specific_timer.pit.present_yes.offset_utc.no_stress.test_specific_timer: PASS (52.11 s)
(33/83) type_specific.ydu-libvirt.timer_management.specific_timer.pit.present_yes.offset_timezone.no_stress.test_specific_timer: PASS (51.96 s)
(34/83) type_specific.ydu-libvirt.timer_management.specific_timer.pit.present_yes.offset_variable.no_stress.test_specific_timer: PASS (52.03 s)
(35/83) type_specific.ydu-libvirt.timer_management.specific_timer.pit.present_yes.offset_localtime.no_stress.test_specific_timer: PASS (52.66 s)
(36/83) type_specific.ydu-libvirt.timer_management.specific_timer.rtc.present_no.offset_utc.no_stress.test_specific_timer: PASS (51.92 s)
(37/83) type_specific.ydu-libvirt.timer_management.specific_timer.rtc.present_no.offset_timezone.no_stress.test_specific_timer: PASS (52.91 s)
(38/83) type_specific.ydu-libvirt.timer_management.specific_timer.rtc.present_no.offset_variable.no_stress.test_specific_timer: PASS (52.10 s)
(39/83) type_specific.ydu-libvirt.timer_management.specific_timer.rtc.present_no.offset_localtime.no_stress.test_specific_timer: PASS (52.94 s)
(40/83) type_specific.ydu-libvirt.timer_management.specific_timer.rtc.present_yes.offset_utc.no_stress.test_specific_timer: PASS (53.08 s)
(41/83) type_specific.ydu-libvirt.timer_management.specific_timer.rtc.present_yes.offset_timezone.no_stress.test_specific_timer: PASS (53.32 s)
(42/83) type_specific.ydu-libvirt.timer_management.specific_timer.rtc.present_yes.offset_variable.no_stress.test_specific_timer: PASS (53.04 s)
(43/83) type_specific.ydu-libvirt.timer_management.specific_timer.rtc.present_yes.offset_localtime.no_stress.test_specific_timer: PASS (52.92 s)
(44/83) type_specific.ydu-libvirt.timer_management.specific_timer.multi_timers.pit_rtc.present_no.offset_utc.no_stress.test_specific_timer: PASS (53.61 s)
(45/83) type_specific.ydu-libvirt.timer_management.specific_timer.multi_timers.pit_rtc.present_no.offset_timezone.no_stress.test_specific_timer: PASS (60.56 s)
(46/83) type_specific.ydu-libvirt.timer_management.specific_timer.multi_timers.pit_rtc.present_no.offset_variable.no_stress.test_specific_timer: PASS (58.10 s)
(47/83) type_specific.ydu-libvirt.timer_management.specific_timer.multi_timers.pit_rtc.present_no.offset_localtime.no_stress.test_specific_timer: PASS (60.72 s)
(48/83) type_specific.ydu-libvirt.timer_management.specific_timer.multi_timers.pit_rtc.present_yes.offset_utc.no_stress.test_specific_timer: PASS (58.66 s)
(49/83) type_specific.ydu-libvirt.timer_management.specific_timer.multi_timers.pit_rtc.present_yes.offset_timezone.no_stress.test_specific_timer: PASS (57.69 s)
(50/83) type_specific.ydu-libvirt.timer_management.specific_timer.multi_timers.pit_rtc.present_yes.offset_variable.no_stress.test_specific_timer: PASS (60.71 s)
(51/83) type_specific.ydu-libvirt.timer_management.specific_timer.multi_timers.pit_rtc.present_yes.offset_localtime.no_stress.test_specific_timer: PASS (58.91 s)
(52/83) type_specific.ydu-libvirt.timer_management.specific_timer.multi_timers.pit_rtc.present_mix.offset_utc.no_stress.test_specific_timer: PASS (65.29 s)
(53/83) type_specific.ydu-libvirt.timer_management.specific_timer.multi_timers.pit_rtc.present_mix.offset_timezone.no_stress.test_specific_timer: PASS (58.77 s)
(54/83) type_specific.ydu-libvirt.timer_management.specific_timer.multi_timers.pit_rtc.present_mix.offset_variable.no_stress.test_specific_timer: PASS (60.85 s)
(55/83) type_specific.ydu-libvirt.timer_management.specific_timer.multi_timers.pit_rtc.present_mix.offset_localtime.no_stress.test_specific_timer: PASS (54.70 s)
(56/83) type_specific.ydu-libvirt.timer_management.specific_timer.multi_timers.hpet_tsc.present_no.offset_utc.no_stress.test_specific_timer: PASS (33.03 s)
(57/83) type_specific.ydu-libvirt.timer_management.specific_timer.multi_timers.hpet_tsc.present_no.offset_timezone.no_stress.test_specific_timer: PASS (37.28 s)
(58/83) type_specific.ydu-libvirt.timer_management.specific_timer.multi_timers.hpet_tsc.present_no.offset_variable.no_stress.test_specific_timer: PASS (37.40 s)
(59/83) type_specific.ydu-libvirt.timer_management.specific_timer.multi_timers.hpet_tsc.present_no.offset_localtime.no_stress.test_specific_timer: PASS (37.27 s)
(60/83) type_specific.ydu-libvirt.timer_management.specific_timer.multi_timers.hpet_tsc.present_yes.offset_utc.no_stress.test_specific_timer: PASS (37.15 s)
(61/83) type_specific.ydu-libvirt.timer_management.specific_timer.multi_timers.hpet_tsc.present_yes.offset_timezone.no_stress.test_specific_timer: PASS (37.04 s)
(62/83) type_specific.ydu-libvirt.timer_management.specific_timer.multi_timers.hpet_tsc.present_yes.offset_variable.no_stress.test_specific_timer: PASS (37.24 s)
(63/83) type_specific.ydu-libvirt.timer_management.specific_timer.multi_timers.hpet_tsc.present_yes.offset_localtime.no_stress.test_specific_timer: PASS (37.49 s)
(64/83) type_specific.ydu-libvirt.timer_management.specific_timer.multi_timers.hpet_tsc.present_mix.offset_utc.no_stress.test_specific_timer: PASS (37.23 s)
(65/83) type_specific.ydu-libvirt.timer_management.specific_timer.multi_timers.hpet_tsc.present_mix.offset_timezone.no_stress.test_specific_timer: PASS (37.11 s)
(66/83) type_specific.ydu-libvirt.timer_management.specific_timer.multi_timers.hpet_tsc.present_mix.offset_variable.no_stress.test_specific_timer: PASS (37.56 s)
(67/83) type_specific.ydu-libvirt.timer_management.specific_timer.multi_timers.hpet_tsc.present_mix.offset_localtime.no_stress.test_specific_timer: PASS (37.17 s)
(68/83) type_specific.ydu-libvirt.timer_management.specific_timer.tsc.present_no.offset_utc.no_stress.test_specific_timer: PASS (35.47 s)
(69/83) type_specific.ydu-libvirt.timer_management.specific_timer.tsc.present_no.offset_timezone.no_stress.test_specific_timer: PASS (35.50 s)
(70/83) type_specific.ydu-libvirt.timer_management.specific_timer.tsc.present_no.offset_variable.no_stress.test_specific_timer: PASS (35.40 s)
(71/83) type_specific.ydu-libvirt.timer_management.specific_timer.tsc.present_no.offset_localtime.no_stress.test_specific_timer: PASS (36.64 s)
(72/83) type_specific.ydu-libvirt.timer_management.specific_timer.tsc.present_yes.offset_utc.no_stress.test_specific_timer: PASS (35.49 s)
(73/83) type_specific.ydu-libvirt.timer_management.specific_timer.tsc.present_yes.offset_timezone.no_stress.test_specific_timer: PASS (35.42 s)
(74/83) type_specific.ydu-libvirt.timer_management.specific_timer.tsc.present_yes.offset_variable.no_stress.test_specific_timer: PASS (35.60 s)
(75/83) type_specific.ydu-libvirt.timer_management.specific_timer.tsc.present_yes.offset_localtime.no_stress.test_specific_timer: PASS (35.75 s)
(76/83) type_specific.ydu-libvirt.timer_management.specific_timer.platform.present_no.offset_utc.no_stress.test_specific_timer: PASS (35.54 s)
(77/83) type_specific.ydu-libvirt.timer_management.specific_timer.platform.present_no.offset_timezone.no_stress.test_specific_timer: PASS (35.40 s)
(78/83) type_specific.ydu-libvirt.timer_management.specific_timer.platform.present_no.offset_variable.no_stress.test_specific_timer: PASS (35.50 s)
(79/83) type_specific.ydu-libvirt.timer_management.specific_timer.platform.present_no.offset_localtime.no_stress.test_specific_timer: PASS (35.49 s)
(80/83) type_specific.ydu-libvirt.timer_management.specific_timer.platform.present_yes.offset_utc.no_stress.test_specific_timer: PASS (35.41 s)
(81/83) type_specific.ydu-libvirt.timer_management.specific_timer.platform.present_yes.offset_timezone.no_stress.test_specific_timer: PASS (35.54 s)
(82/83) type_specific.ydu-libvirt.timer_management.specific_timer.platform.present_yes.offset_variable.no_stress.test_specific_timer: PASS (65.02 s)
(83/83) type_specific.ydu-libvirt.timer_management.specific_timer.platform.present_yes.offset_localtime.no_stress.test_specific_timer: PASS (33.78 s)

File: libvirt/tests/src/timer_management.py [hunk 1]
```diff
@@ -214,23 +213,21 @@ def set_vm_timezone(vm, timezone="America/New_York", windows=False):
...
-    return None
+    numeric_tz = None
+    try:
+        cmd = "TZ='%s'" % tz_name
+        cmd += " date +%z"
+        z = utils.system_output(cmd)
+        numeric_tz = float(z[0:-2]) + float(z[0] + z[-2:]) / 60
+    except Exception, e:
+        logging.error("Convert timezone error: %s", e)
+    return numeric_tz
```
Comment (PandaWei) [inline libvirt/tests/src/timer_management.py]: Great Job!
I'm confused this problem for a long time.
ACK

Comment (PandaWei) [inline libvirt/tests/src/timer_management.py]: Maybe you can make it as another patch and submit it to virttest/utils_xxx

**Comment by waynesun09** [comment]
ACK and merging now, about the function mentioned by @PandaWei , we can add and update it later. Thanks.
